src/devices/machine/qvt102.cpp

```diff
@@ -347,53 +292,74 @@ INPUT_PORTS_END
 // -----21- unused
 // -------0 speaker
 
-READ8_MEMBER(qvt102_state::mcu_bus_r)
+READ8_MEMBER(qvt102_device::mcu_bus_r)
 {
 	return m_kbd_bus;
 }
 
-WRITE8_MEMBER(qvt102_state::mcu_bus_w)
+WRITE8_MEMBER(qvt102_device::mcu_bus_w)
 {
 	m_speaker->level_w(BIT(data, 0));
 	m_kbd_bus = data;
 }
 
-READ_LINE_MEMBER(qvt102_state::mcu_t0_r)
+READ_LINE_MEMBER(qvt102_device::mcu_t0_r)
 {
 	int state = 1;
 
-	if (BIT(m_kbd_bus, 4) == 0) state = BIT(m_keys_special->read(), 0);
-	if (BIT(m_kbd_bus, 5) == 0) state = BIT(m_keys_special->read(), 1);
-	if (BIT(m_kbd_bus, 6) == 0) state = BIT(m_keys_special->read(), 2);
+	// The keyboard firmware also scans for a key at bit 3, and it appears
+	// to be a modifier key and is passed to the host in bit 3 of the
+	// second code, but the terminal firmware appears to ignores it.
+
+	if (BIT(m_kbd_bus, 4) == 0) state &= BIT(m_keys_special->read(), 0);
+	if (BIT(m_kbd_bus, 5) == 0) state &= BIT(m_keys_special->read(), 1);
+	if (BIT(m_kbd_bus, 6) == 0) state &= BIT(m_keys_special->read(), 2);
```

Comment:
According to schematics, SHIFT, CTRL and CAPS are the only things connected here.  Perhaps bit 3 is leftover test code, or something that's present on different hardware?  Also, schematics confirm there's no priority logic, and the `&=` is correct here.